monai/transforms/compose.py

```diff
@@ -41,7 +41,84 @@
 
 logger = get_logger(__name__)
 
-__all__ = ["Compose", "OneOf", "RandomOrder", "SomeOf", "execute_compose"]
+__all__ = [
+    "Compose",
+    "OneOf",
+    "RandomOrder",
+    "SomeOf",
+    "compose_iterator",
+    "ranged_compose_iterator",
+    "execute_compose",
+]
+
+
+def compose_iterator(compose, step_into_all=False):
+    """
+    ``compose_iterator`` is a function that returns an iterator over the transforms in a ``Compose``
+    instance.
+
+    It works by iterating over transforms, and stepping into nested ``Compose`` instances in a
+    recursive fashion. Given the following example:
+
+    ..code-block:: python
+
+    ::
+        t1, t2, t3, t4 = # some transforms
+        c = Compose([t1, Compose([t2, t3]), t4])
+        print([tx for tx in compose_iterator(c)])
+
+        > [t1, t2, t3, t4]
```

Comment:
It looks like the `compose_iterator` returns the flattened transforms, which may ignore the `lazy` arg in the nested `Compose`?

Reply:
It's also true for `overrides` and anything else that might be set on a Compose. I'm looking into a nice way to handle this